lib/warnings.js

```diff
@@ -21,4 +21,6 @@ warning.create('FastifyDeprecation', 'FSTDEP004', 'You are using the legacy preP
 
 warning.create('FastifyDeprecation', 'FSTDEP005', 'You are accessing the deprecated "request.connection" property. Use "request.socket" instead.')
 
+warning.create('FastifyDeprecation', 'FSTDEP006', 'You are decorating Request/Reply with a reference type. This reference is shared amongst all requests. Use onRequest hook instead. Property: %s')
```

Comment:
Should this warning link to this issue?

Reply:
A link to the documentation might be more appropriate.

Reply:
Linking to our documentation can be problematic due to the versioning of it. How do we link to the appropriate version in the PR when we haven't published the content of this PR under a version yet?

Reply:
I don't think this should include a link.